detox/android/detox/src/main/java/com/wix/detox/espresso/AnimatedModuleIdlingResource.java

```diff
@@ -82,91 +83,102 @@ public boolean isIdleNow() {
-                    .field(FIELD_UPDATED_NODES)
-                    .call(METHOD_SIZE)
-                    .get();
-            if (updatedNodesSize > 0) {
-                animationsConsideredIdle = false;
             } else {
-                Object activeAnims = Reflect.on(nodesManager)
-                        .field(FIELD_ACTIVE_ANIMATIONS)
-                        .get();
-                int activeAnimsSize = Reflect.on(activeAnims)
-                        .call(METHOD_SIZE)
-                        .get();
-                if (activeAnimsSize == 0) {
-                    animationsConsideredIdle = true;
-                } else {
-                    animationsConsideredIdle = true;
-                    for (int i = 0; i < activeAnimsSize; ++i) {
-                        int iterations = Reflect.on(activeAnims)
-                                .call(METHOD_VALUE_AT, i)
-                                .field(FIELD_ITERATIONS)
-                                .get();
-                        // -1 means it is looped
-                        if (iterations != -1) {
-                            animationsConsideredIdle = false;
-                            break;
-                        }
-                    }
+                if (isIdleRNOld(animModuleClass)) {
+                    return true;
                 }
             }
 
-            if (operationsAreEmpty && animationsConsideredIdle) {
-                if (callback != null) {
-                    callback.onTransitionToIdle();
-                }
-                // Log.i(LOG_TAG, "AnimatedModule is idle.");
-                return true;
-            }
-
             Log.i(LOG_TAG, "AnimatedModule is busy.");
             Choreographer.getInstance().postFrameCallback(this);
             return false;
         } catch (ReflectException e) {
-            // Log.e(LOG_TAG, "Couldn't set up RN AnimatedModule listener, old RN version?");
-            // Log.e(LOG_TAG, "Can't set up RN AnimatedModule listener", e.getCause());
+            Log.e(LOG_TAG, "Couldn't set up RN AnimatedModule listener, old RN version?");
+            Log.e(LOG_TAG, "Can't set up RN AnimatedModule listener", e.getCause());
         }
 
         if (callback != null) {
             callback.onTransitionToIdle();
         }
+        Log.i(LOG_TAG, "AnimatedModule is idle.");
         return true;
     }
 
+    private boolean isIdleRN51(Object animModuleClass) {
+        Object animModule = Reflect.on(reactContext).call(METHOD_GET_NATIVE_MODULE, animModuleClass).get();
+        Object nodesManager = Reflect.on(animModule).call("getNodesManager").get();
+        boolean hasActiveAnimations = Reflect.on(nodesManager).call("hasActiveAnimations").get();
+        if (!hasActiveAnimations) {
+            if (callback != null) {
+                callback.onTransitionToIdle();
+            }
+            Log.i(LOG_TAG, "AnimatedModule is idle, no operations");
```

Comment:
I think we shouldn't log out idle states. Only busy states.

Originally, I also logged both states, but after the initial debugging it become overwhelming.

Reply:
Agreed